plenum/server/domain_req_handler.py

```diff
@@ -163,32 +164,44 @@ def getNymDetails(state, nym, isCommitted: bool=True):
     def nym_to_state_key(nym: str) -> bytes:
         return sha256(nym.encode()).digest()
 
-    def make_proof(self, path, head_hash=None):
+    def get_value_from_state(self, path, head_hash=None, with_proof=False):
         '''
-        Creates a state proof for the given path in state trie.
-        Returns None if there is no BLS multi-signature for the given state (it can
-        be the case for txns added before multi-signature support).
-
+        Get a value (and proof optionally)for the given path in state trie.
+        Does not return the proof is there is no aggregate signature for it.
         :param path: the path generate a state proof for
+        :param head_hash: the root to create the proof against
+        :param get_value: whether to return the value
         :return: a state proof or None
         '''
         root_hash = head_hash if head_hash else self.state.committedHeadHash
         encoded_root_hash = state_roots_serializer.serialize(bytes(root_hash))
 
-        multi_sig = self.bls_store.get(encoded_root_hash)
-        if not multi_sig:
-            return None
-
-        proof = self.state.generate_state_proof(key=path,
-                                                root=self.state.get_head_by_hash(
-                                                    root_hash),
-                                                serialize=True)
-        encoded_proof = proof_nodes_serializer.serialize(proof)
-        return {
-            ROOT_HASH: encoded_root_hash,
-            MULTI_SIGNATURE: multi_sig.as_dict(),
-            PROOF_NODES: encoded_proof
-        }
+        if with_proof:
```

Comment:
It' better to have 
```
if not with_proof:
    return self.state.get_for_root_hash(root_hash, path), None
...
```
to avoid too many if-else conditions

Reply:
Is it processed?

Reply:
Yes